sdk/cosmos/azure-spring-data-cosmos-test/src/test/java/com/azure/spring/data/cosmos/core/CosmosTemplateIT.java

```diff
@@ -708,6 +709,56 @@ public void createDatabaseWithThroughput() throws ClassNotFoundException {
         assertEquals(expectedRequestUnits, response.getProperties().getManualThroughput());
     }
 
+    @Test
+    public void queryWithMaxDegreeOfParallelism() throws ClassNotFoundException {
+        final String configuredMDPDbName = TestConstants.DB_NAME + "-max-degree-parallel";
+        deleteDatabaseIfExists(configuredMDPDbName);
```

Comment:
Since we are just checking for the same value through `config` object again, we really don't need to create these databases and collections. These are heavy operations and will take extra resources from compute and time.

Reply:
What we should really check here is through some sort of reflection on the CosmosQueryRequestOptions object being passed in the query, and basically checking what is the value of `MaxDegreeOfParallelism`

Reply:
Done.